docs/reference/inference/service-cohere.asciidoc

```diff
@@ -0,0 +1,195 @@
+[[infer-service-cohere]]
+=== Cohere {infer} service
+
+Creates an {infer} endpoint to perform an {infer} task with the `cohere` service.
+
+
+[discrete]
+[[infer-service-cohere-api-request]]
+==== {api-request-title}
+
+`PUT /_inference/<task_type>/<inference_id>`
+
+[discrete]
+[[infer-service-cohere-api-path-params]]
+==== {api-path-parms-title}
+
+`<inference_id>`::
+(Required, string)
+include::inference-shared.asciidoc[tag=inference-id]
+
+`<task_type>`::
+(Required, string)
+include::inference-shared.asciidoc[tag=task-type]
++
+--
+Available task types:
+
+* `completion`,
+* `rerank`,
+* `text_embedding`.
+--
+
+[discrete]
+[[infer-service-cohere-api-request-body]]
+==== {api-request-body-title}
+
+`service`::
+(Required, string)
+The type of service supported for the specified task type. In this case, 
+`cohere`.
+
+`service_settings`::
+(Required, object)
+include::inference-shared.asciidoc[tag=service-settings]
++
+--
+These settings are specific to the `cohere` service.
+--
+
+`api_key`:::
+(Required, string)
+A valid API key of your Cohere account.
+You can find your Cohere API keys or you can create a new one
+https://dashboard.cohere.com/api-keys[on the API keys settings page].
++
+--
+include::inference-shared.asciidoc[tag=api-key-admonition]
+--
+
+`rate_limit`:::
+(Optional, object)
+By default, the `cohere` service sets the number of requests allowed per minute to `10000`.
+This value is the same for all task types.
+This helps to minimize the number of rate limit errors returned from Cohere.
+To modify this, set the `requests_per_minute` setting of this object in your service settings:
++
+--
+include::inference-shared.asciidoc[tag=request-per-minute-example]
+
+More information about Cohere's rate limits can be found in https://docs.cohere.com/docs/going-live#production-key-specifications[Cohere's production key docs].
+--
++
+.`service_settings` for the `completion` task type
+[%collapsible%closed]
+=====
+
+=====
++
+.`service_settings` for the `rerank` task type
+[%collapsible%closed]
+=====
+`model_id`::
+(Optional, string)
+The name of the model to use for the {infer} task.
+To review the available `rerank` models, refer to the
+https://docs.cohere.com/reference/rerank-1[Cohere docs].
+=====
++
+.`service_settings` for the `text_embedding` task type
+[%collapsible%closed]
+=====
+`embedding_type`:::
+(Optional, string)
+Specifies the types of embeddings you want to get back.
+Defaults to `float`.
+Valid values are:
+* `byte`: use it for signed int8 embeddings (this is a synonym of `int8`).
+* `float`: use it for the default float embeddings.
+* `int8`: use it for signed int8 embeddings.
+
+`model_id`::
```

Comment:
Did we also want to include the `similarity` field here?

similarity: Optional, Similarity measure, one of cosine, dot_product, l2_norm. Defaults based on the embedding_type (float -> dot product, int8/byte -> cosine).

Also looks like this is indented a little too far
![image](https://github.com/elastic/elasticsearch/assets/56361221/9ec85dd9-4021-4171-bc87-0e3d8731ef1a)